content/authoring/kata/floating-point.md

```diff
@@ -0,0 +1,277 @@
+It's not always possible, and not always easy, but some categories of problems can be solved in a way which does not introduce intermediate inaccuracies. Staying away from floating-point values is usually a good way to solve problems related to time units (hours, minutes, seconds), angles (expressed as degrees, angle minutes, and angle seconds), monetary values (dollars and cents), or any other mixed units of different magnitudes (for example meters and centimeters).
+
+
+### Know how floating-point values and operations work in your language
+
+Some languages, expecially dynamically typed ones, are particularly susceptible to problems related to floating-point values. For example, JavaScript does not use integers at all, and all arithemtic calculations operate on floating-point numbers. Everything should be good as long as input, output, and intermediate, values stay accurate or do not exceed the value of `Number.MAX_SAFE_INTEGER`, but due to some bug this can happen, and when it happens, it can be difficult to spot.
+
+#### Overflow
+
+One of such problems is overflow in Javascript. For example, consider a kata with following task: _"Given two natural numbers `a` and `b`, calculate and return their least common multiple."_ Random tests are careful enough to generate such values of `a` and `b`, which always should give the resulting LCM as less than `Number.MAX_SAFE_INTEGER`. However, the reference solution uses following implementation:
+
+```javascript
+function lcm(a, b) {
+    return a * b / gcd(a, b); //gcd is a helper function to calculate greatest common divisor
+}
+```
+
+Can you see where potential problem is? Even though `a`, `b`, and final result are guaranteed to be less than `Number.MAX_SAFE_INTEGER`, intermediate value of `a * b` can overflow, resulting in incorrect result being returned. Now reference solution has a bug!
+
+You need to make sure that your reference solution is correct and can handle all test inputs which will be generated and fed to it. Otherwise, correctness of tests depends on the favor of random number generator, and some valid solutions might fail the tests in an unpredictable, and difficult to debug way.
+
+#### Division
+
+In languages like JavaScript or Python, floating-point values can be introduced by mistake when performing division. JavaScript has no integer division operator, and operation of `a / b` always results in floating-point values. If you are interested in integer division, you need to remember to convert the result to integer by yourself. Python has two division operators: `a // b` for integer division (or, "real floor division"), and `a / b` for "real" division. Both operators can be easily confused, and `/` will convert integers to floats and return a float.
+
+
+### Use relative comparisons and avoid strict equality
+
+Have you ever encountered a problem that some solution does not pass because tests expect a slightly different value, even though you used right formula?
```

Comment:
```suggestion
Have you ever encountered a problem that some solution does not pass because tests expect a slightly different value, even though you used the right formula?
```